src/ex4/server/services/itemManagerService.js

```diff
@@ -0,0 +1,117 @@
+const fs = require("fs");
+const path = require("path");
+const itemFile = "./server/data/itemsList.json";
+const cacheDir = "./server/data/cache";
+const cacheFilePath = "./server/data/cache/cache.json";
+const cacheFileName = "cache.json";
+const autoDeleteCacheService = require("./autoDeleteCacheService");
+
+const { v4: ideKeyGen } = require("uuid");
+
+async function checkIfPokemonIdInCache(pokemonId) {
+  let cacheData = [];
+  try {
+    if (!fs.existsSync(cacheFilePath)) {
+      createCacheFile(cacheFilePath);
+
+      cacheWriteAutoDelete(pokemonId, cacheData);
+
+      return false;
+    }
+
+    cacheData = JSON.parse(fs.readFileSync(cacheFilePath));
```

Comment:
`JSON.parse` should get string input, here you pass it a `Buffer`
should change to `fs.readFileSync(cacheFilePath).toString()`